src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -111,13 +111,15 @@ public override string AssemblyQualifiedName
 
         public override Type BaseType => _typeSymbol.BaseType!.AsType(_metadataLoadContext);
 
+        public override Type DeclaringType => _typeSymbol.ContainingType?.ConstructedFrom.AsType(_metadataLoadContext);
+
         private string? _fullName;
 
         public override string FullName
         {
             get
             {
-                if (_fullName == null)
+                if (_fullName == null && !IsGenericParameter)
```

Comment:
same question, is null expected when `_fullName == null && IsGenericParameter`?

Reply:
Yes, null is expected here. Sample:
```
Type genericParameterType = typeof(List<>).GetGenericArguments()[0];
Console.WriteLine($"Name: {genericParameterType.Name}");
Console.WriteLine($"IsGenericParameter: {genericParameterType.IsGenericParameter}");
Console.WriteLine($"FullName: {genericParameterType.FullName ?? "null"}");
Console.WriteLine($"AssemblyQualifiedName: {genericParameterType.AssemblyQualifiedName ?? "null"}");
```
output
```
Name: T
IsGenericParameter: True
FullName: null
AssemblyQualifiedName: null
```